pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -205,8 +213,32 @@ protected void map(LongWritable key, Text value, Context context) throws IOExcep
     }
 
     protected void setSegmentNameGenerator(SegmentGeneratorConfig segmentGeneratorConfig, Integer seqId,
-        Path hdfsAvroPath, File dataPath) {
+                                           Path hdfsAvroPath, File dataPath) {
+      Schema schema = segmentGeneratorConfig.getSchema();
+
+      if(schema == null ||
+        schema.getTimeFieldSpec() == null ||
+        schema.getTimeFieldSpec().getIncomingGranularitySpec() == null){
+        return;
+      }
+
+      String timeFormatStr = schema.getTimeFieldSpec().getIncomingGranularitySpec().getTimeFormat();
+
+      if (timeFormatStr.equals(TimeGranularitySpec.TimeFormat.EPOCH.toString())) {
+        return;
+      }
+
+      String simpleDateTypeStr = timeFormatStr.split(":")[1];
 
+      // Throws illegal argument exception if invalid
+      try{
+        new SimpleDateFormat(simpleDateTypeStr);
+      }
+      catch (IllegalArgumentException e) {
+        throw new RuntimeException("Could not parse simple date format " + simpleDateTypeStr, e);
+      }
+      segmentGeneratorConfig.setSimpleDateFormat(simpleDateTypeStr);
```

Comment:
can you refactor this line and the next into a separate method and comment why you have paired millis/with the simple date format, provide an example so outside pinot users can also follow this?